atomic_reactor/utils/operator.py

```diff
@@ -304,14 +487,34 @@ def __init__(self, path, data, pullspec_heuristic=default_pullspec_heuristic):
             implementation will be used. For more information, see the
             default_pullspec_heuristic() function in this module.
         """
+        self.check_csv(path, data)
+        self.path = path
+        self.data = data
+        self._pullspec_heuristic = pullspec_heuristic
+
+    @classmethod
+    def check_csv(cls, path, data):
         try:
             if data.get("kind") != OPERATOR_CSV_KIND:
                 raise NotOperatorCSV("Not a ClusterServiceVersion")
         except AttributeError:
             raise NotOperatorCSV("File does not contain a YAML object")
-        self.path = path
-        self.data = data
-        self._pullspec_heuristic = pullspec_heuristic
+        # Validate CSVs against schema before *any* other processing
```

Comment:
Please use common method for validation we use elsewhere, or refactor it as needed

Reply:
Which common method? `osbs-client:read_yaml()`? How will I be able to use that when it requires a path to a schema file, and I have a schema embedded in a Python dictionary?

Reply:
Split current read_yaml into two methods (I dunno why it was not done as that before as those things are orthogonal)


```python
def validate_with_schema(data_as_py_object, schema_as_dict):
    # validation code from current read_yaml, raisess exceptions if data are invalid
   ....
   pass

def read_yaml(....):
    # load files, first part of current read_yaml
    ....
    data = ...
    schema = ...
    validate_with_schema(data, schema)
    return data
```

Reply:
ACK, opened https://github.com/containerbuildsystem/osbs-client/pull/1009